### PR TITLE
#KTOR-936 Migrate to explicit api mode

### PR DESCRIPTION
**Pull request by e5l**
Fixes [`#KTOR-936`](https://youtrack.jetbrains.com/issue/KTOR-936)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -160,11 +167,13 @@ allprojects {
         }
     }
 
-
-
     configurations { testOutput }
 
     kotlin {
+        if (!ignoreExplicitApi.contains(project.name)){
+            explicitApi()
```
Comment (cy6erGn0m) [inline build.gradle:174]: So it becomes error, right?

Comment (e5l) [reply build.gradle:174]: Yep

File: build.gradle [hunk 2]
```diff
@@ -99,6 +99,13 @@ def check(Object version, String libVersion, String libName) {
     }
 }
 
+def ignoreExplicitApi = [
+    "ktor-client-tests",
+    "ktor-client-json-tests",
+    "ktor-server-test-host",
```
Comment (cy6erGn0m) [inline build.gradle:105]: Why is the test host excluded?

Comment (e5l) [reply build.gradle:105]: To exclude suites

File: build.gradle [hunk 3]
```diff
@@ -99,6 +99,14 @@ def check(Object version, String libVersion, String libName) {
     }
 }
 
+def ignoreExplicitApi = [
```
Comment (Stexxe) [inline build.gradle:102]: This name is better suited for a method, not for a list of things. I suggest  `DisabledExplicitApiModeProjects` or something shorter.

File: build.gradle [hunk 4]
```diff
@@ -247,7 +253,7 @@ if (build_snapshot_train) {
                     include 'META-INF/MANIFEST.MF'
                 }.getFiles().first()
 
-                manifest.readLines().each {
+                manifes.readLines().each {
```
Comment (Stexxe) [inline build.gradle:256]: Typo?

**Review by Stexxe — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 4 above)
Comment (e5l) [reply build.gradle:256]: yep. thanks for catching

**Review by Stexxe — APPROVED** [review]

**Pull request merged**